### PR TITLE
Improve sampler documentation, use doctest

### PR DESCRIPTION
**Pull request by guillaumekln**
# Description

Documentation improvement.

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

### CODE REVIEW

File: tensorflow_addons/seq2seq/sampler.py [hunk 1]
```diff
@@ -29,25 +29,34 @@
...
-        (finished, cell_input, cell_state) = sampler.next_inputs(
-            time_step, cell_output, cell_state, sample_ids)
-        if tf.reduce_all(finished):
-            break
-    ```
-
-    Note that the input_tensors should not be fed to the Sampler as __init__()
-    parameters. Instead, they should be fed by decoders via initialize().
+    Sampler classes implement the logic of sampling from the decoder output distribution
+    and producing the inputs for the next decoding step. In most cases, they should not be
```
Comment (Harsh188) [inline tensorflow_addons/seq2seq/sampler.py:33]: and produce?

**Review by Harsh188 — APPROVED** [review]

File: tensorflow_addons/seq2seq/sampler.py (hunk 1 above)
Comment (guillaumekln) [reply tensorflow_addons/seq2seq/sampler.py:33]: I think the structure is correct: "the logic of sampling and producing". But I'm not a native English speaker so not 100% sure.

Comment (Harsh188) [reply tensorflow_addons/seq2seq/sampler.py:33]: Mmm, I see. It was just a little confusing when I first read it.

**Comment by AakashKumarNain** [comment]
Thanks @guillaumekln for fixing it. LGTM. Waiting for any additional inputs from @qlzh727

**Review by qlzh727 — APPROVED** [review]
